File: Javascript/MergeTwoBinaryTrees.js
function TreeNode(val,left,right){
    this.val = (val === undefined) ? 0 : val
    this.left = (left === undefined) ? 0 : left
    this.right = (right === undefined) ? 0 : right
}

var mergeBinaryTrees = function (root1,root2){

    function helper(node1,node2){

        if(node1 === null && node2 === null){
            return null
        }

        const v1 = (node1 !== null) ? node1.val:0
        const v2 = (node2 !== null) ? node2.val:0

        const root = new TreeNode(v1+v2)

        root.left = helper((node1 !== null) ? node1.left:null , 
        (node2 !== null) ? node2.left:null)
        root.right = helper((node1 !== null) ? node1.right:null , 
        (node2 !== null) ? node2.right:null)

        return root
    }

    return helper(root1,root2)
}

File: Python/MergeTwoBinaryTrees.py
class TreeNode:
    def __init__(self,val=0,left=None,right=None) -> None:
        self.val = val
        self.left = left
        self.right = right


class Solution:
    def mergeBinaryTrees(root1,root2):

        def helper(node1,node2):
            if not node1 and not node2:
                return None
            
            v1 = node1.val if node1 else 0
            v2 = node2.val if node2 else 0

            root = TreeNode(v1+v2)

            root.left = helper(node1.left if node1 else None,node2.left if node2 else None)
            root.right = helper(node1.right if node1 else None,node2.right if node2 else None)

            return root
        
        return helper(root1,root2)